shoulder/generator/cxx_header_generator.py

```diff
@@ -48,350 +46,403 @@ def generate(self, objects, outpath):
-        accessor = accessor.format(
-            indent = self._indent_string(),
-            size_t = reg_cxx_size,
-            func = config.is_bit_cleared_function,
-            regname = reg_cxx_name,
-            msb = field.msb
-        )
-        outfile.write(accessor)
-
-        # Check bit disabled from an integer value
-        accessor = "{indent}inline {size_t} {func}({size_t} {arg}) noexcept "
-        accessor += "{{ IS_BIT_DISABLED_FUNC({arg}, {msb}) }}\n"
-        accessor = accessor.format(
-            indent = self._indent_string(),
-            size_t = reg_cxx_size,
-            func = config.is_bit_cleared_function,
-            arg = reg_val_cxx_name,
-            msb = field.msb
-        )
-        outfile.write(accessor)
-
-        # Enable the bit in the system register directly
-        accessor = "{indent}inline void {func}() noexcept "
-        accessor += "{{ SET_SYSREG_BITS_BY_MASK_FUNC({regname}, {mask}) }}\n"
-        accessor = accessor.format(
-            indent = self._indent_string(),
-            func = config.bit_set_function,
-            regname = reg_cxx_name,
-            mask = mask
-        )
-        outfile.write(accessor)
-
-        # Enable the bit in an integer value
-        accessor = "{indent}inline {size_t} {func}({size_t} {arg}) noexcept "
-        accessor += "{{ SET_BITS_BY_MASK_FUNC({arg}, {mask}) }}\n"
-        accessor = accessor.format(
-            indent = self._indent_string(),
-            size_t = reg_cxx_size,
-            func = config.bit_set_function,
-            arg = reg_val_cxx_name,
-            mask = mask
-        )
-        outfile.write(accessor)
-
-        # Disable the bit in the system register directly
-        accessor = "{indent}inline void {func}() noexcept "
-        accessor += "{{ CLEAR_SYSREG_BITS_BY_MASK_FUNC({regname}, {mask}) }}\n"
-        accessor = accessor.format(
-            indent = self._indent_string(),
-            func = config.bit_clear_function,
-            regname = reg_cxx_name,
-            mask = mask,
-        )
-        outfile.write(accessor)
-
-        # Disable the bit in an integer value
-        accessor = "{indent}inline {size_t} {func}({size_t} {arg}) noexcept "
-        accessor += "{{ CLEAR_BITS_BY_MASK_FUNC({arg}, {mask}) }}\n"
-        accessor = accessor.format(
-            indent = self._indent_string(),
-            size_t = reg_cxx_size,
-            func = config.bit_clear_function,
-            arg = reg_val_cxx_name,
-            mask = mask
-        )
-        outfile.write(accessor)
+                if(field.msb == field.lsb):
+                    self._generate_bitfield_accessors(outfile, reg, field)
+
+                else:
+                    self._generate_field_accessors(outfile, reg, field)
 
-    def _generate_field_accessors(self, reg, field, outfile):
+            _in_field_namespace(self, outfile)
+
+    def _generate_field_constants(self, outfile, reg, field):
```

Comment:
I'm a big fan of using these constant definitions instead of hardcoding it into the function. It makes it very easy to go to that field and know the critical information for it quickly.